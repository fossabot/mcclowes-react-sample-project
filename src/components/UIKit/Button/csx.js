import styled from "styled-components";

export const ButtonWrapper = styled.button`
  background: ${props => props.theme.colors.primary};
  border-radius: 3px;
  border: none;
  color: ${props => props.theme.colors.white};
  cursor: pointer;
  font-size: 14px;
  padding: 0.5em 1em;
  border: 2px solid ${props => props.theme.colors.primary};
  display: flex;
  justify-items: center;
  align-items: center;

  &:hover {
    opacity: 0.75;
  }

  &:active {
    opacity: 0.5;
  }

  &:focus {
    outline: none;
    border: 2px solid ${props => props.theme.colors.alt};
  }

  ${props => (props.fill ? "width: 100%;" : "")};

  ${props => (props.large ? `padding: 0.75em 1.5em;` : "")};

  svg {
    margin-right: 0.5em;
  }
`;
