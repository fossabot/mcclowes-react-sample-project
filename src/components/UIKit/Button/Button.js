import React from "react";
import { ButtonWrapper } from "./csx";

const Button = props => {
  const { children, text, to = "/", doClick } = props;

  return (
    <ButtonWrapper onClick={doClick} to={to}>
      {text}
      {children}
    </ButtonWrapper>
  );
};

export default Button;
