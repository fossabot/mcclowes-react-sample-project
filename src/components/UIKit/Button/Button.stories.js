import React from "react";
import Button from "./Button";

export default { title: "Button" };

export const empty = () => {
  return <Button />;
};

export const withText = () => {
  return <Button>Hello World</Button>;
};
