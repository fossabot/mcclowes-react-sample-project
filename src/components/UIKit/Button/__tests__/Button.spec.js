import Button from "../Button";
import React from "react";
import { shallow } from "enzyme";

describe("Button", () => {
  it("renders component", () => {
    const wrapper = shallow(<Button text="Default Button" />);

    expect(wrapper).toMatchSnapshot();
  });

  it("hovers", () => {
    const wrapper = shallow(<Button text="Default Button" />);

    wrapper.simulate("hover");

    expect(wrapper).toMatchSnapshot();
  });

  describe.skip("to", () => {
    it("default", () => {
      const wrapper = shallow(<Button text="Default Button" />);

      expect(wrapper.find("ButtonWrapper").prop("to")).toBe("/");
    });

    it("passed", () => {
      const wrapper = shallow(<Button text="Default Button" to="/some/url" />);

      expect(wrapper.find("ButtonWrapper").prop("to")).toBe("/some/url");
    });
  });
});
