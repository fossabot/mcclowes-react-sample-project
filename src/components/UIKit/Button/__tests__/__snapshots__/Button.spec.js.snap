// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders component 1`] = `
<csx__ButtonWrapper
  theme={
    Object {
      "colors": Object {
        "alt": "#888",
        "primary": "#aaa",
      },
      "sizes": Object {
        "border": Object {
          "radius": "3px",
        },
      },
    }
  }
  to="/"
>
  Default Button
</csx__ButtonWrapper>
`;
